src/Pages/Applications/DevtronApps/Details/AppConfigurations/Navigation/Navigation.helper.tsx

```diff
@@ -0,0 +1,175 @@
+import { NavLink, generatePath } from 'react-router-dom'
+
+import { ReactComponent as Lock } from '@Icons/ic-locked.svg'
+import { ReactComponent as ProtectedIcon } from '@Icons/ic-shield-protect-fill.svg'
+import { ReactComponent as ICStamp } from '@Icons/ic-stamp.svg'
+import { ReactComponent as ICEditFile } from '@Icons/ic-edit-file.svg'
+import { URLS } from '@Config/routes'
+import { CollapsibleListItem } from '@Pages/Shared/CollapsibleList'
+import { ResourceConfigStage, ResourceConfigState } from '@Pages/Applications/DevtronApps/service.types'
+
+import {
+    CustomNavItemsType,
+    EnvConfigRouteParams,
+    EnvConfigType,
+    EnvironmentOptionType,
+    EnvResourceType,
+    ExtendedCollapsibleListItem,
+} from '../AppConfig.types'
+import { BASE_CONFIGURATIONS } from '../AppConfig.constants'
+
+export const renderNavItem = (item: CustomNavItemsType, isBaseConfigProtected?: boolean) => {
+    const linkDataTestName = item.title.toLowerCase().split(' ').join('-')
+
+    return (
+        <NavLink
+            data-testid={`${linkDataTestName}-link`}
+            key={item.title}
+            onClick={(event) => {
+                if (item.isLocked) {
+                    event.preventDefault()
+                }
+            }}
+            className="dc__nav-item cursor fs-13 lh-32 cn-7 w-100 br-4 px-8 flexbox dc__align-items-center dc__content-space dc__no-decor"
+            to={item.href}
+        >
+            <span className="dc__truncate nav-text">{item.title}</span>
+            {item.isLocked && (
```

Comment:
We can extract the icon in separate method and add if else block to make it more readable